doc/syslog/Syslog_Source_IP_test_plan.md

```diff
@@ -0,0 +1,165 @@
+
+
+## Overview
+
+SSIP is a feature which allows user to change UDP packet source IP address.
+Any configured address can be used for source IP mangling.
+This might be useful for security reasons.
+
+SSIP also extends the existing syslog implementation with VRF device and server UDP port configuration support. The feature doesn't change the existing DB schema which makes it fully backward compatible.
+
+### Scope
+
+The test is to verify SSIP take effect after doing relevant SSIP configuration   
+
+### Scale / Performance
+
+No scale/performance test involved in this test plan
+
+### Related **DUT** CLI commands
+```
+User interface:
+
+config
+|--- syslog
+     |--- add <server_ip> OPTIONS
+     |--- del <server_ip>
+
+show
+|--- syslog
+
+Options:
+
+config syslog add
+
+-s|--source - source ip address
+-p|--port - server udp port
+-r|--vrf - vrf device
+
+```
+
+### Supported topology
+The test will be supported on t0.
+
+
+### Test cases #1 -  Configure syslog server with VRF/Source:unset/unset
+1. Configure syslog server with VRF/Source:unset/unset like below:
+```
+config syslog add '2.2.2.2' 
+```
+2. Check syslog config by show syslog, the result should like below:
+```
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2     N/A     N/A     N/A
+```
+3. Check the corresponding interface of rsyslog server can receive the syslog message with port 514
+4. Del syslog server config like below:
+```
+config syslog del '2.2.2.2' 
+```
+5. Check syslog config by show syslog, the relevant config should be removed
+6. Check the corresponding interface of rsyslog server can not receive any syslog message with port 514
+7. Repeat steps 1-5 by setting different port, and check the port of received syslog message will be changed to the set one
+
+
+### Test cases #2 -  Configure syslog server with VRF/Source: unset/set
+1. Configure syslog server with VRF/Source: unset/set like below:
+```
+config syslog add '2.2.2.2' --source "1.1.1.1"
+```
+2. Check syslog config by show syslog, the result should like below:
+```
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2     1.1.1.1     N/A     N/A
+```
+3. Check the corresponding interface of rsyslog server can receive the syslog message with port 514 and src ip 1.1.1.1
+4. Del syslog server config like below:
+```
+config syslog del '2.2.2.2' 
+```
+5. Check syslog config by show syslog, the relevant config should be removed
+6. Check the corresponding interface of rsyslog server can not receive any syslog message with port 514 and src ip 1.1.1.1
+7. Repeat steps 1-6 by setting different port, and check the port of received syslog message will be changed to the set one
+
+
+### Test cases #3 -  Configure syslog server with VRF/Source: set/unset
+1. For vrf: Default, mgmt, Vrf-data, Configure syslog server with VRF/Source: set/unset like below:
+```
+config syslog add '2.2.2.2' --vrf "Default"
```

Comment:
@JibinBao the keyword is `default` - please fix a typo

Reply:
will fix it. thanks